powerbi-visual-builder/src_visual/visual.ts

```diff
@@ -148,15 +148,101 @@ namespace powerbi.extensibility.visual {
             }
           }
         }
+        for (const v of categories) {
+          if (
+            v.source.roles[column.powerBIName] &&
+            !columnToValues[column.powerBIName]
+          ) {
+            columnToValues[column.powerBIName] = {
+              values: CharticulatorContainer.Dataset.convertColumnType(
+                v.values.map(x => (x == null ? null : x.valueOf())),
+                column.type
+              ),
+              highlights: v.values.map((value, i) => {
+                return false;
+              })
+            };
+            found = true;
+          }
+        }
         if (!found) {
           return null;
         }
       }
 
+      const linkColumns = options.dataViews[0].categorical.categories;
+      const links =
+        this.template.tables[1] &&
+        (this.template.tables[1].columns as PowerBIColumn[]);
+
+      if (links && linkColumns) {
+        for (const column of links) {
+          for (const v of linkColumns) {

```

Comment:
This logic looks almost identical to the logic above at line 153ish.  I think it makes sense to pull the logic out into a function.